cylc/flow/scripts/install.py

```diff
@@ -60,15 +174,31 @@ def main(parser, options, reg):
                 exc
             ) from None
 
+    source_dir, rundir, _flow_name = install_workflow(
+        flow_name=opts.flow_name,
+        source=opts.source,
+        run_name=opts.run_name,
+        no_run_name=opts.no_run_name,
+        no_symlinks=opts.no_symlinks
+    )
+
     for entry_point in pkg_resources.iter_entry_points(
         'cylc.post_install'
     ):
         try:
-            entry_point.resolve()(
-                dir_=os.getcwd(),
-                opts=options,
-                dest_root=get_suite_run_dir(suite)
-            )
+            if entry_point.name == 'rose_files':
+                entry_point.resolve()(
+                    dir_=source_dir,
+                    opts=opts,
+                    dest_root=str(rundir)
+                )
+            elif entry_point.name == 'rose_opts':
+                entry_point.resolve()(
+                    dest_path=str(rundir),
+                    opt_conf_keys=opts.opt_confs,
+                    defines=opts.defines,
+                    suite_defines=opts.define_suites
+                )
```

Comment:
Just pass the opts straight through so all plugins have the same interface. The plugins can extract what they need from this.

```suggestion
            entry_point.resolve()(
                dir_=source_dir,
                opts=opts,
                dest_root=str(rundir)
            )
```